plugins/node/opentelemetry-instrumentation-ioredis/README.md

```diff
@@ -0,0 +1,88 @@
+# OpenTelemetry ioredis Instrumentation for Node.js
+
+[![Gitter chat][gitter-image]][gitter-url]
+[![dependencies][dependencies-image]][dependencies-url]
+[![devDependencies][devDependencies-image]][devDependencies-url]
+[![Apache License][license-image]][license-image]
+
+This module provides automatic instrumentation for [`ioredis`](https://github.com/luin/ioredis).
+
+For automatic instrumentation see the
+[@opentelemetry/node](https://github.com/open-telemetry/opentelemetry-js/tree/master/packages/opentelemetry-node) package.
+
+## Installation
+
+```sh
+npm install --save @opentelemetry/instrumentation-ioredis
+```
+
+### Supported Versions
+
+- `>=2.0.0`
+
+## Usage
+
+To load a specific instrumentation (**ioredis** in this case), specify it in the Node Tracer's configuration
+
+```javascript
+const { NodeTracerProvider } = require('@opentelemetry/node');
+const { IORedisInstrumentation } = require('@opentelemetry/instrumentation-ioredis');
+
+const provider = new NodeTracerProvider({
+  // be sure to disable old plugin
+  plugins: {
```

Comment:
you won't be able to load instrumentation using node plugin loader, please use `registerInstrumentations`

Reply:
I copied this over from e.g. https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-instrumentation-http/README.md#usage
It just shows that old plugin should be disabled and not how to load the instrumentation.

The instrumentation works by just creating it via `new IORedisInstrumentation()` as shown a few lines below.

Reply:
Yeah currently all the instrumentation on the core have this inside their readme, most users still use plugins anyway. @obecny Is the new loading system ready ? I didn't see any PR to update documentation on the core

Reply:
New loading system is ready yet. I'm finishing removing PluginLoader from node tracer. If you don't want to add `registerInstrumentations` you should still show how to set a trace provider for the instrumentation and enable it so user knows how to use it - check [graphql](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/master/plugins/node/opentelemetry-instrumentation-graphql). Although the easiest and less code would be to use `registerInstrumentations` flow but it will depends on user.

Reply:
Instrumentations are enabled on default so no need to call `enable()`. Why should we teach users to call `enable()`?

I think the graphql sample is not showing how users should startup their app. First the `TracerProvider` should be created and registered and then the instrumentations should be created as they use the global `TracerProvider` at construction time. Manually setting the `TracerProvider` at several locations seems to be redundant and error prone to me.

There might be special cases where one would have several different `TracerProviders` and `ContextManagers` but this requires special care as there are a lot places in code where e.g. a tracer uses the global context even it was not created by the global `TracerProvider`. In short I would not recommend anyone to go this path.

I don't understand why `registerInstrumentations()` result in less code and is easer then just calling `new FooInstrumentation()` followed by `new BarInstrumentation()`,....

Reply:
Instrumentation is not enabled by default nor the tracer provider or meter provider is set. You have to call all of that manually or you can use `registerInstrumentations` where all of that is happening automatically.

Reply:
> Instrumentation is not enabled by default

seems to be a bug then, see https://github.com/open-telemetry/opentelemetry-js/blob/27a8d4530c56257c191468b073669909bf8892a1/packages/opentelemetry-instrumentation/src/instrumentation.ts#L38-L41


>  nor the tracer provider or meter provider is set.

see https://github.com/open-telemetry/opentelemetry-js/blob/27a8d4530c56257c191468b073669909bf8892a1/packages/opentelemetry-instrumentation/src/instrumentation.ts#L44-L47
A `Tracer` using the current global `TracerProvider` is created, same for `Meter`.

Reply:
hmm I have wrote that, how could I miss that omg, anyway I will fix the enable in the coming PR where I remove the PluginLoader

Reply:
Honestly speaking I find the current implementation by far better. What is the benefit of creating disabled instrumentations - just to require an extra enable step? But most likely better to move this discussion into a separate issue/PR.

Reply:
`What is the benefit of creating disabled instrumentations?` sry what are you referring to ? I have already wrote that I miss that ...

Reply:
If I understood you correct you intend to change `InstrumentationAbstract` to use `enabled: false` as default.
This means every user has to enable them compared to now where they "just work".

Reply:
no I didn't not I was not aware of that I have already added a default tracer and meter and simply missed the enabling plugin - as it is done by `registerIntrumentations`.